### PR TITLE
Fix stage order

### PR DESCRIPTION
**Pull request by maribel4amido**
[[FARM-333 Fix the bug on stage ordering](https://amidodevelopment.atlassian.net/browse/FARM-333) ]

#### 📲 What

Fix ordering of the stages for a crop

#### 🤔 Why
		
Ensure stages for a crop matches the stages on the CMS
		
#### 🛠 How
		
fetch stages for a crop and assign the stages to the crop. unpublished stages are excluded.

#### 👀 See
		
see the APK build link (when completed)
		 
#### ✅ Acceptance criteria

Stage order follows the order on the CMS

#### 🕵️‍♂️ How to test

see the APK build link (when completed)

#### Reviewers

@farmsmart/amido @farmsmart/wamf

### CODE REVIEW

File: lib/data/repositories/plot_repository.dart [hunk 1]
```diff
@@ -16,19 +18,37 @@ class PlotRepository{
...
+  Future<dynamic> _getListOfCropsWithImages(
+      List<CropEntity> cropsWithoutImages) {
     return _firestoreManager.getCropsImagePath(cropsWithoutImages);
   }
 
+  Future<List<CropEntity>> getArticles() async {
+    Stopwatch sw = new Stopwatch();
+    sw.start();
+    List<CropEntity> listOfCrops = await _getListOfCrops();
+    debugPrint('getListOfCrops() ${sw.elapsed.inMilliseconds} ms ');
```
Comment (LeeMatthewHiggins) [inline lib/data/repositories/plot_repository.dart:39]: Debug code and prints ideally should be removed. A nice debug timer function could be useful, we can then add remove quickly. - Firebase test lab also has performance warning tests that run that should pick this up too, you can then link that to the PR build and see when your PR hurts performance.

**Review by LeeMatthewHiggins — APPROVED** [review]
If this is tested let get it in, but we have some debug code in this PR that normally should be removed. (timers and debug prints).

**Comment by LeeMatthewHiggins** [comment]
@jessicahernandezhv @isaacmartinezimv does this create conflicts with any pending PR´s?

**Comment by maribel4amido** [comment]
> @jessicahernandezhv @isaacmartinezimv does this create conflicts with any pending PR´s?

Only one branch needing these fixes in, I'll help Isaac with the merge over slack later.

File: lib/data/repositories/plot_repository.dart (hunk 1 above)
Comment (maribel4amido) [reply lib/data/repositories/plot_repository.dart:39]: Agree. For beta1 though I think these are safe to go in.

Comment (raubreysmith) [reply lib/data/repositories/plot_repository.dart:39]: more concerned about the stopwatch assuming it runs outside of the debug build, however the using of printing should be made clear in the standards guide if not already @albertmiro

**Review by LeeMatthewHiggins — COMMENTED** [review]
the only thing is this does change the repo api, so could conflict with others.

**Review by raubreysmith — APPROVED** [review]
Exploration coverage
- adding/removing stages
- setting status to draft
- no stages
- basic usage of articles / related articles load times

Notes
- Updates only happen if you navigate to a different tab, not when you view the details of a crop
- performance is still good
- placeholder image doesn't render with the container so you get container >delay > placeholder > delay > crop image

**Pull request merged**